### DISCUSSION TITLE
Digital Signatures for RefProp DLL's?

### DISCUSSION THREAD

**Discussion by wnewhauser** [usnistgov/REFPROP-wrappers#627] [Ideas]
Would it be possible to add digital signatures to REFPROP.DLL and REFPRP64.DLL? When referenced from a C# program, the anti-virus is wanting to block the C# program from running. I'm speculating it's because the anti-virus sees references to a native-code dll that is not digitally signed and thinks it could be malicious. Not 100% sure this would fix the problem, but it would be a step in the right direction. Vendors like Comodo and others offer code signing certs fairly inexpensively. Thoughts...

**Comment by ianhbell** [comment]
Yes! Currently we sign the installer but not the DLL. It would be easy enough to sign the DLL. What version of the DLL are you looking for right now?

**Reply by wnewhauser** [reply]
10.0.

Also, would be awesome if you could put the version # in the DLL's so you it when you right click in Windows Explorer and select Properties->Details you see it. Thanks!

**Reply by ianhbell** [reply]
I added that here: https://github.com/usnistgov/REFPROP-cmake/commit/b5be54bc54beccff05adf65b733c249c0e05aae8

**Comment by ianhbell** [comment]
Do you happen to know how to add the version to a shared library in CMake?

**Comment by ianhbell** [comment]
To answer my own question: https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html .  Now to grep out the version from the REFPROP sources...

**Comment by wnewhauser** [comment]
Any idea when you could publish the signed RefProp DLL's? Have a current project where the anti-virus won't let our own program execute  presumably because it's linked to the unsigned RefProp DLLs. TIA!

**Reply by ianhbell** [reply]
I forwarded this request to our engineer responsible for signing
